From 8a78e454f93c07c4cabd6434bec6e33c1e8f968b Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 29 Jul 2015 10:02:07 +0530
Subject: [PATCH] [EM/Charger]: Remove unwanted code from the charger(bq24192)
 driver

In charger(bq24192) driver, the functions slave mode enable/disable
charging is defined and which is not used anywhere. Also in the
suspend case not required to do enable/disable charging. Hence
this unnecessary changes are removed.

Change-Id: I1b3541a2eede29bf9476fa15ac6ac13381f666a0
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-14896
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/power/bq24192_charger.c | 58 +----------------------------------------
 1 file changed, 1 insertion(+), 57 deletions(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index 1ea80119d8a6..aee4722fd820 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -2058,48 +2058,6 @@ static inline int register_otg_vbus(struct bq24192_chip *chip)
 	return 0;
 }
 
-int bq24192_slave_mode_enable_charging(int volt, int cur, int ilim)
-{
-	struct bq24192_chip *chip = i2c_get_clientdata(bq24192_client);
-	int ret;
-
-	mutex_lock(&chip->event_lock);
-	chip->inlmt = ilim;
-	if (chip->inlmt >= 0)
-		bq24192_set_inlmt(chip, chip->inlmt);
-	mutex_unlock(&chip->event_lock);
-
-	chip->cc = chrg_cur_to_reg(cur);
-	if (chip->cc)
-		bq24192_set_cc(chip, chip->cc);
-
-	chip->cv = chrg_volt_to_reg(volt);
-	if (chip->cv)
-		bq24192_set_cv(chip, chip->cv);
-
-	mutex_lock(&chip->event_lock);
-	ret = bq24192_enable_charging(chip, true);
-	if (ret < 0)
-		dev_err(&chip->client->dev, "charge enable failed\n");
-
-	mutex_unlock(&chip->event_lock);
-	return ret;
-}
-EXPORT_SYMBOL(bq24192_slave_mode_enable_charging);
-
-int bq24192_slave_mode_disable_charging(void)
-{
-	struct bq24192_chip *chip = i2c_get_clientdata(bq24192_client);
-	int ret;
-
-	mutex_lock(&chip->event_lock);
-	ret = bq24192_enable_charging(chip, false);
-	if (ret < 0)
-		dev_err(&chip->client->dev, "charge enable failed\n");
-
-	mutex_unlock(&chip->event_lock);
-	return ret;
-}
 static int bq24192_get_chip_version(struct bq24192_chip *chip)
 {
 	int ret;
@@ -2484,24 +2442,10 @@ static int bq24192_remove(struct i2c_client *client)
 #ifdef CONFIG_PM
 static int bq24192_suspend(struct device *dev)
 {
-	int ret;
 	struct bq24192_chip *chip = dev_get_drvdata(dev);
 
 	dev_dbg(&chip->client->dev, "bq24192 suspend\n");
-	if (chip->is_charging_enabled) {
-		mutex_lock(&chip->event_lock);
-		ret = bq24192_enable_charging(chip, true);
-		if (ret < 0)
-			dev_err(&chip->client->dev, "charging enable failed\n");
-		mutex_unlock(&chip->event_lock);
-	} else {
-		mutex_lock(&chip->event_lock);
-		ret = bq24192_enable_charging(chip, false);
-		if (ret < 0)
-			dev_err(&chip->client->dev, "charging disable failed\n");
-		mutex_unlock(&chip->event_lock);
-	}
-	return ret;
+	return 0;
 }
 
 static int bq24192_resume(struct device *dev)
-- 
1.9.1

